test/general/lambda_naming.pass.cpp

```diff
@@ -58,7 +60,13 @@ int main() {
     EXPECT_TRUE(!is_equal, "wrong return value from equal");
     auto does_1_exist = ::std::find(policy, buf_begin, buf_end, 1);
     EXPECT_TRUE(does_1_exist - buf_begin == 1000, "wrong return value from find");
-#endif
+#endif // !_ONEDPL_FPGA_DEVICE
+
+#elif !__SYCL_UNNAMED_LAMBDA__ && _ONEDPL_BACKEND_SYCL
+    // ::std::for_each(policy, buf_begin, buf_end, [](int& x) { x++; }); // It's not allowed. Policy with different name is needed
+    ::std::for_each(oneapi::dpl::execution::make_device_policy<class ForEach>(policy), buf_begin, buf_end, [](int& x) { x++; });
+    auto red_val = ::std::reduce(policy, buf_begin, buf_end, 1);
+    EXPECT_TRUE(red_val == 1002, "wrong return value from reduce");
```

Comment:
I would expect `2001`, because the buffer size is `n` and it contains only `2`s after applying of `for_each` above.

Reply:
Correct. Thanks!